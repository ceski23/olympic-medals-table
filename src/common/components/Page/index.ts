export { Page } from "./Page";
export type { CountryMedalsData } from "./Page";
