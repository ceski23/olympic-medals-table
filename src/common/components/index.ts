export * from "./Select";
export * from "./Input";
export * from "./Page";
export * from "./App";
