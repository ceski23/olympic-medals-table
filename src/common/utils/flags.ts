export const flags = Object.fromEntries(
  Object.entries(
    import.meta.glob("../../../node_modules/country-flag-icons/3x2/*.svg", {
      as: "url",
      eager: true,
    })
  ).map(([id, url]) => [id.match(/([A-Z][A-Z])\.svg/)?.[1] ?? id, url])
);
