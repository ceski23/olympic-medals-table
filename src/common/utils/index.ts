export * from "./guards";
export * from "./styles";
export * from "./flags";
