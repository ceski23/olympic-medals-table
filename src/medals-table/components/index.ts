export * from "./CountryFlag";
export * from "./MedalsTable";
export * from "./SortingIndicator";
export * from "./Table";
export * from "./CountryForm";
