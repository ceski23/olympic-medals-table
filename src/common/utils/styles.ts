import { Theme } from "@emotion/react";
import { Breakpoint } from "common/styles";

export const mediaQueryUp =
  (breakpoint: Breakpoint) =>
  ({ theme }: { theme: Theme }) =>
    `@media (min-width: ${theme.breakpoints[breakpoint] + 0.01}px)`;

export const mediaQueryUpTo =
  (breakpoint: Breakpoint) =>
  ({ theme }: { theme: Theme }) =>
    `@media (max-width: ${theme.breakpoints[breakpoint]}px)`;
