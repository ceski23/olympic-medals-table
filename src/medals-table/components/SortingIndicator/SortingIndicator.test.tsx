import { SortingIndicator } from "medals-table/components";
import { renderWithProviders } from "test/utils";

describe("SortingIndicator", () => {
  it("should render correctly without direction", () => {
    const { asFragment } = renderWithProviders(
      <SortingIndicator direction={false} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render correctly with ascending direction", () => {
    const { asFragment } = renderWithProviders(
      <SortingIndicator direction="asc" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render correctly with descending direction", () => {
    const { asFragment } = renderWithProviders(
      <SortingIndicator direction="desc" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
