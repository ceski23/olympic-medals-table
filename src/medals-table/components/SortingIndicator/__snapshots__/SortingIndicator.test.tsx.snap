// Vitest Snapshot v1

exports[`SortingIndicator > should render correctly with ascending direction 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 20px;
  height: 20px;
}

<svg
    class="emotion-0 emotion-1"
    viewBox="0 0 312.85 385.74"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18.26 174.61H294.6c16.23 0 24.355-19.617 12.879-31.094L169.309 5.336c-7.113-7.113-18.645-7.113-25.762 0L5.377 143.516C-6.092 154.993 2.033 174.61 18.26 174.61z"
      fill="#707070"
    />
    <path
      d="M5.38 242.23l138.17 138.18c7.113 7.11 18.648 7.11 25.762 0l138.17-138.18c11.477-11.473 3.352-31.094-12.879-31.094H18.253c-16.227 0-24.352 19.621-12.879 31.094z"
      fill="#dddddd"
    />
  </svg>
</DocumentFragment>
`;

exports[`SortingIndicator > should render correctly with descending direction 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 20px;
  height: 20px;
}

<svg
    class="emotion-0 emotion-1"
    viewBox="0 0 312.85 385.74"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18.26 174.61H294.6c16.23 0 24.355-19.617 12.879-31.094L169.309 5.336c-7.113-7.113-18.645-7.113-25.762 0L5.377 143.516C-6.092 154.993 2.033 174.61 18.26 174.61z"
      fill="#dddddd"
    />
    <path
      d="M5.38 242.23l138.17 138.18c7.113 7.11 18.648 7.11 25.762 0l138.17-138.18c11.477-11.473 3.352-31.094-12.879-31.094H18.253c-16.227 0-24.352 19.621-12.879 31.094z"
      fill="#707070"
    />
  </svg>
</DocumentFragment>
`;

exports[`SortingIndicator > should render correctly without direction 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 20px;
  height: 20px;
}

<svg
    class="emotion-0 emotion-1"
    viewBox="0 0 312.85 385.74"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18.26 174.61H294.6c16.23 0 24.355-19.617 12.879-31.094L169.309 5.336c-7.113-7.113-18.645-7.113-25.762 0L5.377 143.516C-6.092 154.993 2.033 174.61 18.26 174.61z"
      fill="#dddddd"
    />
    <path
      d="M5.38 242.23l138.17 138.18c7.113 7.11 18.648 7.11 25.762 0l138.17-138.18c11.477-11.473 3.352-31.094-12.879-31.094H18.253c-16.227 0-24.352 19.621-12.879 31.094z"
      fill="#dddddd"
    />
  </svg>
</DocumentFragment>
`;
